airflow/providers_manager.py

```diff
@@ -950,6 +957,14 @@ def _discover_auth_backends(self) -> None:
                     if _sanity_check(provider_package, auth_backend_module_name + ".init_app", provider):
                         self._api_auth_backend_module_names.add(auth_backend_module_name)
 
```

Comment:
Out of scope, but this file looks like it is begging for a helper method. :thinking:

Reply:
Of course, it's not as easy as it looks.   I tried [this](https://github.com/aws-mwaa/upstream-to-airflow/commit/6ad3e78d0d7205cd1e63821c5087282ff66ac4c1) but it broke a couple dozen unit tests.  Maybe not worth it.

Reply:
Yeah. Likely a nice refactor would be nice here. But separately from this one.